parsing_input.c

```diff
@@ -0,0 +1,61 @@
+#include "shell.h"
+/**
+ * calculate_tokens - function to calculate token command
+ * @lineptr: pointer to string
+ * @delim: delimiter between strings
+ * Return: number of tokens
+ */
+int calculate_tokens(char *lineptr, char *delim)
+{
+	int counter = 0;
+	char *token, *line_copy;
+
+	line_copy = _strcpy(line_copy, lineptr);
+	token = strtok(line_copy, delim);
+	while (token != NULL)
+	{
+		counter++;
+		token = strtok(line_copy, delim);
+	}
+	return (counter);
+}
+/**
+ * _parse_to_token - tokenize the input (command)
+ * @num_token: number of token returned by the function calculate_token
+ * @lineptr: pointer to the command to be tokenized
+ * @delim: the delimiter between tokens
+ * Return: array of tokens to be passed to the execve function
+ */
+char **_parse_to_token(int num_token, char *lineptr, char *delim)
+{
+	char **arr_t;
+	int i = 0;
+	char *line_copy, *token;
+
+	line_copy = _strdup(lineptr);
+	if (line_copy == NULL)
+	{
+		free(lineptr);
+		exit(EXIT_FAILURE);
+	}
+	arr_t = malloc(sizeof(char *) * (num_token + 1));
```

Comment:
you did not initialise the variable num_token, i.e. num_token = calculate_tokens(lineptr, delim);